state/executor.go

```diff
@@ -253,6 +253,38 @@ func NewTransition(config chain.ForksInTime, snap Snapshot, radix *Txn) *Transit
 	}
 }
 
+func (t *Transition) WithStateOverride(override types.StateOverride) error {
+	for addr, o := range override {
+		if o.Nonce != nil {
+			t.state.SetNonce(addr, *o.Nonce)
+		}
+
+		if o.Balance != nil {
+			t.state.SetBalance(addr, o.Balance)
+		}
+
+		if o.Code != nil {
+			t.state.SetCode(addr, o.Code)
+		}
+
+		if o.State != nil && o.StateDiff != nil {
+			return fmt.Errorf("cannot override both state and state diff")
+		}
+
+		if o.State != nil {
+			t.state.SetFullStorage(addr, o.State)
+		}
+
+		if o.StateDiff != nil {
```

Comment:
this check is not needed, range operator will work with nil maps and slices

Reply:
Done in https://github.com/0xPolygon/polygon-edge/pull/1337/commits/788a5857600d952439f5d4b0b0f2fa16b13c5de7